### PR TITLE
Cleanup and stabilization for TUT tracker

### PR DESCRIPTION
**Pull request by predicative**
This is the first round of changes to the TUT tracker (`srnn_tracker`) extracted from the publicly released code at `dev/tut-tracker-prr-20200716`. The main addition is basic homography-based stabilization, but there's also some reformatting, refactoring, and bug fixes.

The consumer-visible changes are then:
- `srnn_tracker` now has an optional input port `homography_src_to_ref` that it uses to do tracking in a stabilized coordinate space if present
- Tracks should no longer acquire two new states in a single frame under certain circumstances
- The first-pass IoU tracking should now work as intended

### CODE REVIEW

**Comment by linus-sherrill** [comment]
Is this ready to go? I'm not the best to review all the python changes, so could somebody else approve.

**Comment by linus-sherrill** [comment]
Is this ready to merge? I am not the best reviewer for the python changes.

**Comment by dstoup** [comment]
I added some python reviewers so we can get good feedback.

**Review by tao558 — CHANGES_REQUESTED** [review]

File: python/kwiver/sprokit/processes/pytorch/utils/track.py [hunk 1]
```diff
@@ -32,12 +32,15 @@
 import collections
 
 class track_state(object):
```
[the remark this answers is not in the document]
Comment (neal-siekierski) [reply python/kwiver/sprokit/processes/pytorch/utils/track.py:34]: The underlying concepts are certainly similar, but what's done here isn't trivially rewritable in terms of the `vital` types as I understand it. #1075 goes into more detail about some of what might need to happen and also applies to the current state of `master`.

**Review by tao558 — APPROVED** [review]

**Comment by tao558** [comment]
Approved but I'd recommend still getting input from the other Python folks

File: python/kwiver/sprokit/processes/pytorch/utils/iou_tracker.py [hunk 2]
```diff
@@ -39,22 +39,30 @@ def _track_iou(self, track_set, track_state_list):
...
+                    continue
+                match, iou = nonreject_dets[0]
+                if iou < self._iou_accept_threshold:
+                    continue
+
+                # Add the matching det
+                track.updated_flag = True
+                track_set.update_track(track.track_id, match)
+                # remove matching detection from detections
+                del track_state_list[track_state_list.index(match)]
```
Comment (Erotemic) [inline python/kwiver/sprokit/processes/pytorch/utils/iou_tracker.py:65]: `track_state_list.remove(match)` is faster and more readable. 

Benchmark for evidence in favor of my faster claim:

```python
        import random
        max_list_size = 1000
        max_item_size = int(2 ** 32)

        import timerit
        ti = timerit.Timerit(1000, bestof=10, verbose=2)

        rng = random.Random(0)
        for timer in ti.reset('using remove'):
            list_size = rng.randint(1, max_list_size)
            items = [rng.randint(0, max_item_size) for _ in range(list_size)]
            remove_index = rng.randint(0, list_size - 1)
            remove_item = items[remove_index]
            with timer:
                items.remove(remove_item)

        rng = random.Random(0)
        for timer in ti.reset('using del'):
            list_size = rng.randint(1, max_list_size)
            items = [rng.randint(0, max_item_size) for _ in range(list_size)]
            remove_index = rng.randint(0, list_size - 1)
            remove_item = items[remove_index]
            with timer:
                del items[items.index(remove_item)]
```

Results:

```
Timed using remove for: 1000 loops, best of 10
    time per loop: best=0.646 µs, mean=1.216 ± 0.4 µs
Timed using del for: 1000 loops, best of 10
    time per loop: best=0.745 µs, mean=1.300 ± 0.5 µs
```

File: python/kwiver/sprokit/processes/pytorch/srnn_tracker.py [hunk 3]
```diff
@@ -289,6 +352,10 @@ def timing(desc, f):
             in_img_c = self.grab_input_using_trait('image')
             timestamp = self.grab_input_using_trait('timestamp')
             dos_ptr = self.grab_input_using_trait('detected_object_set')
+            if self.has_input_port_edge('homography_src_to_ref'):
+                homog_f2f = self.grab_input_using_trait('homography_src_to_ref')
+            else:
+                homog_f2f = None
             print('timestamp =', repr(timestamp))
```
Comment (Erotemic) [inline python/kwiver/sprokit/processes/pytorch/srnn_tracker.py:359]: Should this print statement be a logging statement?

File: python/kwiver/sprokit/processes/pytorch/utils/models.py [hunk 4]
```diff
@@ -222,47 +222,35 @@ def __init__(self, app_model='', motion_model='', interaction_model='', bbox_mod
...
+
+            """
             if use_gpu_flag:
-                self.appearance = AppearanceLSTM().cuda()
-            else:
-                self.appearance = AppearanceLSTM()
-            if app_model:
-                snapshot = torch.load(app_model)
-                self.appearance.load_state_dict(snapshot['state_dict'])
+                model = model.cuda()
```
Comment (Erotemic) [inline python/kwiver/sprokit/processes/pytorch/utils/models.py:231]: Using `.cuda` makes a lot of assumptions. This should at least be `.to(device)` even if device is hard-coded to 0. What would be even better is if the device was able to be specified.

**Review by Erotemic — COMMENTED** [review]
Looks pretty reasonable to me. The code seems reasonably clean. I mainly verified high-level items, I didn't look in depth at the logic.

**Pull request closed without merging**